### PR TITLE
Variable Data Format Enhancements

### PR DESCRIPTION
**Pull request by marcoAntonioNina**
- Enable UI Variable
- enable inspector

https://github.com/ProcessMaker/processmaker/issues/2001

### CODE REVIEW

File: src/form-builder-controls.js [hunk 1]
```diff
@@ -61,41 +62,171 @@ const colorProperty = {
...
+  type: 'FormInput',
+  field: 'validation',
+  panel: 'variable',
+  config: {
+    label: 'Validation',
+    name: 'validation',
+    helper: 'Validation rules ensure the integrity and validity of the data.',
+  },
+};
+const ReadOnly = {
```
Comment (chipit24) [inline src/form-builder-controls.js]: I like that this was broken out into a reusable object 👍 

@velkymx @marcoAntonioNina

There is already a PR open for adding the "read only" property: https://github.com/ProcessMaker/screen-builder/pull/265/files. Not sure why this work was duplicated here.

This also seems to set the `disabled` attribute, not the `readonly` attribute. Bootstrap components accept `readonly`: https://getbootstrap.com/docs/4.0/components/forms/#readonly. Was this intentional, or do we actually want this to disable the component? From the perspective of an average user, the difference may to be very noticeable, and `readonly` only seems to work on text input and textarea.

File: src/form-builder-controls.js [hunk 2]
```diff
@@ -131,15 +262,18 @@ export default [
...
+            validation: 'required',
+            options: [
+              {
+                value: 'normal',
+                content: 'Normal',
+              },
+              {
+                value: 'bold',
+                content: 'Bold',
+              },
```
Comment (chipit24) [inline src/form-builder-controls.js]: If formatting changes are required, they should be added to the `.eslintrc.js` file. A good bit of this PR is unnecessary formatting changes; please have your code editor format based on the ESLint config file.

File: src/components/vue-form-builder.vue [hunk 3]
```diff
@@ -149,68 +149,104 @@
...
       <b-card no-body class="p-0 h-100">
-        <b-card-header>{{ $t('Inspector') }}</b-card-header>
-
+        <b-card-header>
+          {{ $t('Inspector') }}
+        </b-card-header>
         <b-card-body class="p-0 h-100 overflow-auto">
+
           <b-button
-            v-b-toggle.configuration
```
Comment (chipit24) [inline src/components/vue-form-builder.vue]: There is already a PR for this change that just needs the copy updated: https://github.com/ProcessMaker/screen-builder/pull/263. And the implementation here doesn't seem to work as expected (the variable accordion is actually in the "open" state but empty):

Screen Shot 2019-07-02 at 1 51 04 PM

Comment (caleeli) [reply src/components/vue-form-builder.vue]: This requirement comes from this ticket:
https://github.com/ProcessMaker/processmaker/issues/2001

In which requires the read only property must be inside the Variable panel:
image

File: src/form-builder-controls.js (hunk 1 above)
Comment (caleeli) [reply src/form-builder-controls.js]: Following the requirement in the ticket: https://github.com/ProcessMaker/processmaker/issues/2001

The readonly property must be part of the Variable panel

image

File: src/components/vue-form-builder.vue (hunk 3 above)
Comment (chipit24) [reply src/components/vue-form-builder.vue]: I assume there was a mixup and @marcoAntonioNina grabbed https://github.com/ProcessMaker/processmaker/issues/2001 and started doing work in this repo, even though there have already been tickets and PRs created for that issue in this repo.

**Comment by chipit24** [comment]
PRs for adding the "Variable" and "Design" accordions, re-naming "Variable" to "Key Name", and adding the `readonly` prop have been merged. The only thing left that this PR should handle is adding the "Data Type" dropdown:

Screen Shot 2019-07-03 at 1 55 55 PM

**Comment by caleeli** [comment]
Please review #288 instead
